pkg/processor/runtime/rpc/connection/abstract.go

```diff
@@ -327,27 +342,68 @@ func (be *AbstractEventConnection) processItem(item interface{}, functionLogger
 		be.functionLogger = nil
 		return nil, errors.Wrapf(err, "Can't encode item: %+v", item)
 	}
-	processingResults, ok := <-be.resultChan
 
-	// We don't use defer to reset be.functionLogger since it decreases performance
-	be.functionLogger = nil
+	// if eventTimeout is 0, we wait for response without timeout
+	// if it is set, we wait either for response or the timeout
+	if be.connectionManager.GetConfig().eventTimeout == 0 {
+		processingResults, ok := <-be.resultChan
+		// We don't use defer to reset be.functionLogger since it decreases performance
+		be.functionLogger = nil
 
-	if !ok {
-		msg := "Client disconnected"
-		be.Logger.Error(msg)
+		if !ok {
+			msg := "Client disconnected"
+			be.Logger.Error(msg)
 
-		// TODO: support status for socket separately when implementing multiple socket support
-		be.connectionManager.SetStatus(status.Error)
-		return nil, errors.New(msg)
+			// TODO: support status for socket separately when implementing multiple socket support
+			be.connectionManager.SetStatus(status.RestartRequired)
+			return nil, errors.New(msg)
+		}
+		// if processingResults.err is not nil, it means that whole batch processing was failed
+		if processingResults.Err != nil {
+			return nil, processingResults.Err
+		}
+		return processingResults, nil
+	} else {
+		return be.waitForResponseWithTimeout()
 	}
-	// if processingResults.err is not nil, it means that whole batch processing was failed
-	if processingResults.Err != nil {
-		return nil, processingResults.Err
+}
+
+func (be *AbstractEventConnection) waitForResponseWithTimeout() (*result.BatchedResults, error) {
+	ticker := time.NewTicker(be.connectionManager.GetConfig().eventTimeout)
+	defer ticker.Stop()
+
+	select {
+	case processingResults, ok := <-be.resultChan:
+		// We don't use defer to reset be.functionLogger since it decreases performance
+		be.functionLogger = nil
+
+		if !ok {
+			msg := "Client disconnected"
+			be.Logger.Error(msg)
+
+			// TODO: support status for socket separately when implementing multiple socket support
+			be.connectionManager.SetStatus(status.RestartRequired)
+			return nil, errors.New(msg)
+		}
+		// if processingResults.err is not nil, it means that whole batch processing was failed
+		if processingResults.Err != nil {
+			return nil, processingResults.Err
+		}
+		return processingResults, nil
+	case <-ticker.C:
+		be.Logger.WarnWith("Event processing timeout, connection should be restarted")
+		be.status.SetStatus(status.RestartRequired)
+		return &result.BatchedResults{
```

Comment:
Why do you return a `BatchedResults` object?

Reply:
@TomerShor what should I return instead?